app/coffee/Features/Labels/LabelsController.coffee

```diff
@@ -0,0 +1,26 @@
+EditorRealTimeController = require "../Editor/EditorRealTimeController"
+LabelsHandler = require './LabelsHandler'
+logger = require 'logger-sharelatex'
+
+
+module.exports = LabelsController =
+
+	getAllLabels: (req, res, next) ->
+		project_id = req.params.Project_id
+		LabelsHandler.getAllLabelsForProject project_id, (err, projectLabels) ->
+			if err?
+				logger.err {project_id, err}, "[LabelsController] error getting all labels from project"
+				return next(err)
+			res.json {projectId: project_id, projectLabels: projectLabels}
+
+	getLabelsForDoc: (req, res, next) ->
+		project_id = req.params.Project_id
+		doc_id = req.params.doc_id
+		LabelsHandler.getLabelsForDoc project_id, doc_id, (err, docLabels) ->
+			if err?
+				logger.err {project_id, doc_id, err}, "[LabelsController] error getting labels from doc"
+				return next(err)
+			EditorRealTimeController.emitToRoom project_id, 'docLabelsUpdated', {
```

Comment:
Sorry, I think what I meant wasn't clear here. I think it's the parent `getLabelsForDoc` that is not semantically named, and it's odd that we're doing some broadcast of something when all we're doing is GETing. I think the top level HTTP method should be something like `broadcastLabelsForDoc` and should be a post request. The client doesn't actually care about the HTTP response anyway right, because it gets updated through socket io, so it's not even using the GET response at the moment?